src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -289,6 +285,91 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     }
   }
 
+  /**
+    * Collect data from Blocks Table, Operations Table, and the Tezos RPC,
+    * return the appropriate list of VoteAggregates, and write them to the Votes table.
+    * @param ex the needed ExecutionContext to combine multiple database operations
+    * @return a future result of the number of rows stored to db, if supported by the driver
+    */
+  private[this] def processTezosVotes()(implicit ex: ExecutionContext): Future[Option[Int]] = {
+    import cats.instances.list._
+    import cats.instances.future._
+    import cats.syntax.traverse._
+    import cats.implicits._
+    import slick.jdbc.PostgresProfile.api._
+
+    logger.info("Processing latest Tezos votes data...")
+
+    def partitionByVote(fields: Seq[VotingFields]): (Seq[VotingFields], Seq[VotingFields], Seq[VotingFields]) = {
+      val (yays, notYays) = fields.partition(x => x._6 == Some("yay"))
+      val (nays, notYaysOrNays) = notYays.partition(x => x._6 == Some("nay"))
+      val (passes, _) = notYaysOrNays.partition(x => x._6 == Some("pass"))
+      (yays, nays, passes)
+    }
+
+    def calculateStake(fields: Seq[VotingFields]): Future[BigDecimal] =
+      Future
+        .traverse(
+          fields.map {
+            case (_, _, hash, _, _, _, address) =>
+              (BlockHash(hash), AccountId(address.get))
+          }
+        ) { x =>
+          tezosNodeOperator.getAccountBalanceForBlock(x._1, x._2)
```

Comment:
Maybe fusing the steps might simplify?
```scala
Future.traverse(fields){
  case (_, _, hash, _, _, _, address) =>
    tezosNodeOperator.getAccountBalanceForBlock(BlockHash(hash), AccountId(address.gett))
}.map[...]
```

Also, what if `address` is not there? Calling `.get` will make it explode and crash the process.

Better to explicitly consider how to deal with the missing data.

Reply:
Very good points, I'll add them in.